### PR TITLE
Configuration extension for arduino-nano-33-iot

### LINKED ISSUE #16267: Is ADC driver implemented for SAMD21G18A?
The code does not compile if I use the adc.h interface with arduino-nano-33-iot.

#include "periph/adc.h"
adc_init(ADC_LINE(1));

make flash BOARD=arduino-nano-33-iot

undefined reference to `adc_init'
collect2: error: ld returned 1 exit status

If I add in the Makefile:

FEATURES_REQUIRED += periph_adc 

then: There are unsatisfied feature requirements: periph_adc

### PR DESCRIPTION
**Pull request by IchiroKuroki**
### Contribution description
Added missing configuration for ADC / PWM / SPI

### Testing procedure
Tested with:
tests/periph_adc
tests/periph_spi
tests/periph_pwm

### Issues/PRs references
Discussed in #16247 and #16267
Closes #16267

### CODE REVIEW

File: boards/arduino-nano-33-iot/include/periph_conf.h [hunk 1]
```diff
@@ -139,6 +139,72 @@ static const uart_conf_t uart_config[] = {
...
+ * @name PWM configuration
+ * @{
+ */
+#define PWM_0_EN            1
+#define PWM_1_EN            1
+
+#if PWM_0_EN
+/* PWM0 channels */
+static const pwm_conf_chan_t pwm_chan0_config[] = {
+    // GPIO pin, MUX value, TCC channel
```
Comment (fjmolinas) [inline boards/arduino-nano-33-iot/include/periph_conf.h:152]: Can you replace this one by doxygen style comments?

**Comment by fjmolinas** [comment]
Thanks a lot for the contribution @IchiroKuroki, can you possibly post the test output you have? It would be awesome to have some history of your test results that way. Otherwise, the PR looks good, only a minor comment on my side codewise, otherwise regarding your commits I think `boards/arduino-nano-33-iot: ...` would have been more inline with our [contributing guidelines(https://github.com/RIOT-OS/RIOT/blob/master]/CONTRIBUTING.md#commit-conventions), but as it is now its good as well.

**Comment by dylad** [comment]
This PR looks good to me. Configuration matches datasheet and also doc from Arduino.

**Comment by IchiroKuroki** [comment]
> can you possibly post the test output you have

For the ADC, I used the tests/periph_adc test. I connected each ADC pin to GND and then to a 0.88v voltage source.
For each pin I received:

ADC_LINE(num): 0        - GND
ADC_LINE(num): 890    - 0.88v (ADC_REFCTRL_REFSEL_INT1V  and ADC_RES_10BIT)
Unfortunately, I have no way to vary the voltage for the test. 

For the SPI I used tests/periph_spi test. I connected MISO and MOSI and got the following result:

 init 1 1 4 0 21
 send TEST_DATA
 Sent bytes
       0    1    2    3    4    5    6    7    8
     0x54 0x45 0x53 0x54 0x5f 0x44 0x41 0x54 0x41
        T    E    S    T    _    D    A    T    A 
 Received bytes
       0    1    2    3    4    5    6    7    8 
     0x54 0x45 0x53 0x54 0x5f 0x44 0x41 0x54 0x41
        T    E    S    T    _    D    A    T    A

For the PWM I used tests/periph_pwm. I made test configurations, and ran the "osci" command. Then, for each pin, using an oscilloscope, I checked that the frequency is kept at 732Hz, and the duty changes cyclically 0% -> ...-> 100% -> ...-> 0% -> ...
[pwm_test.txt](https://github.com/RIOT-OS/RIOT/files/6300386/pwm_test.txt)

**Comment by fjmolinas** [comment]
Please squash @IchiroKuroki!

**Review by fjmolinas — APPROVED** [review]
ACK!

**Pull request merged**